src/main/java/com/avispl/symphony/dal/device/core510i/QSYSCoreCommunicator.java

```diff
@@ -72,8 +112,36 @@ protected void authenticate() {
 	 * @param controllableProperty This is the property to be controlled
 	 */
 	@Override
-	public void controlProperty(ControllableProperty controllableProperty) {
-		// I am writing
+	public void controlProperty(ControllableProperty controllableProperty) throws IllegalAccessException {
+		String property = controllableProperty.getProperty();
+		String value = String.valueOf(controllableProperty.getValue());
+
+		if (this.logger.isDebugEnabled()) {
+			this.logger.debug("controlProperty property " + property);
+			this.logger.debug("controlProperty value " + value);
+		}
+
+		String[] splitProperty = property.split(String.valueOf(QSYSCoreConstant.HASH));
```

Comment:
I know you are controlling the property via getMultipleStatistics, but if something went wrong in the cloud it send you the wrong property, the adapter shouldn't throw an NPE exception, we know the error and we should throw a specific err msg. I think it would be better to check the property contains HASH. and also check the splitProperty[0], splitProperty[1] are existed or not.

Reply:
Yes, I fixed it, I will check if the length of splitProperty, if it is not equal to 2, I will throw an exception for it.